tool_path_planner/include/tool_path_planner/path_generator.h

```diff
@@ -13,11 +13,15 @@
 
 namespace tool_path_planner
 {
-using ToolPathSegment = std::vector<Eigen::Isometry3d>;
-using ToolPath = std::vector<ToolPathSegment>;
-using ToolPaths = std::vector<ToolPath>;
+/** @brief A set of contiguous waypoints that lie on the same line created by a "slice" through the mesh*/
+using RasterSegment = std::vector<Eigen::Isometry3d, Eigen::aligned_allocator<Eigen::Isometry3d>>;
+/** @brief A set of raster segments that lie on the same line created by a tool path "slice",
+ * but are not connected (e.g. due to a hole in the mesh) */
+using Raster = std::vector<RasterSegment>;
+/** @brief A set of rasters generated by various slices through a mesh */
+using ToolPath = std::vector<Raster>;
```

Comment:
I imagine that much arguing could be done about what the exact names should be, but I think these new names make more sense than the old.  Furthermore, the fact that the new names are clearly documented is an immense improvement.

There may be a legitimate critique of these new names in the field of edge path generation: How do these map onto the arrangment of Eigen::Isometry3d, std::vector<Eigen::Isometry3d>, etc. generated by that planner?  Comments here and/or in the edge path generator may be helpful.

Reply:
I don't think the `PathGenerator` interface should be used for edge paths. To me, edge paths and raster paths are fundamentally different, so they should have different typedefs and interface classes. As we continue our improvements to `noether` I would suggest we separate the two types of planners.

Regarding this PR, it's not intended to be a full out fix. Right now I just want to make it a little more clear what these objects represent

Reply:
> I don't think the `PathGenerator` interface should be used for edge paths. To me, edge paths and raster paths are fundamentally different, so they should have different typedefs and interface classes. As we continue our improvements to `noether` I would suggest we separate the two types of planners.

I agree that the algorithms are fundamental different, but from a user prospective the interface is not. Whether you are generating a surface path versus an edge path you provide a mesh as an input and expect toolpaths in return.